tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/PublicAssetGetMethodContract.scala

```diff
@@ -0,0 +1,318 @@
+import com.linagora.tmail.james.common.probe.PublicAssetProbe
+import com.linagora.tmail.james.jmap.publicAsset.ImageContentType.ImageContentType
+import com.linagora.tmail.james.jmap.publicAsset.{ImageContentType, PublicAsset, PublicAssetId, PublicAssetIdFactory, PublicURI}
+import com.linagora.tmail.james.jmap.settings.JmapSettingsStateFactory
+import eu.timepit.refined.auto._
+import io.netty.handler.codec.http.HttpHeaderNames.ACCEPT
+import io.restassured.RestAssured.{`given`, requestSpecification}
+import io.restassured.http.ContentType.JSON
+import net.javacrumbs.jsonunit.JsonMatchers.jsonEquals
+import org.apache.http.HttpStatus.SC_OK
+import org.apache.james.GuiceJamesServer
+import org.apache.james.jmap.api.model.{IdentityId, Size}
+import org.apache.james.jmap.api.model.Size.Size
+import org.apache.james.jmap.core.AccountId
+import org.apache.james.jmap.core.ResponseObject.SESSION_STATE
+import org.apache.james.jmap.http.UserCredential
+import org.apache.james.jmap.mail.BlobId
+import org.apache.james.jmap.rfc8621.contract.Fixture.{ACCEPT_RFC8621_VERSION_HEADER, ANDRE, ANDRE_PASSWORD, BOB, BOB_PASSWORD, DOMAIN, authScheme, baseRequestSpecBuilder}
+import org.apache.james.mailbox.model.ContentType
+import org.apache.james.utils.DataProbeImpl
+import org.hamcrest.Matchers.hasKey
+import org.junit.jupiter.api.{BeforeEach, Test}
+
+object PublicAssetGetMethodContract {
+  val PUBLIC_ASSET_ID: PublicAssetId = PublicAssetIdFactory.generate()
+  val CONTENT_TYPE: ContentType = ContentType.of("image/png")
+  val IMAGE_CONTENT_TYPE: ImageContentType = ImageContentType.from(CONTENT_TYPE).toOption.get
+  val ASSET_CONTENT: Array[Byte] = Array[Byte](1, 2, 3)
+  val SIZE: Size = Size.sanitizeSize(ASSET_CONTENT.length)
+  val IDENTITY_ID = IdentityId.generate
+  val IDENTITY_IDS: Seq[IdentityId] = Seq(IDENTITY_ID)
+  val CREATION_REQUEST: PublicAsset = PublicAsset(id = PUBLIC_ASSET_ID,
+    publicURI = PublicURI.from(PUBLIC_ASSET_ID, AccountId.from(BOB).toOption.get, new URI("http://localhost:8080/")),
+    size = SIZE,
+    contentType = IMAGE_CONTENT_TYPE,
+    blobId = BlobId("blob1"),
+    identityIds = IDENTITY_IDS)
+}
+
+trait PublicAssetGetMethodContract {
+  @BeforeEach
+  def setUp(server: GuiceJamesServer): Unit = {
+    server.getProbe(classOf[DataProbeImpl])
+      .fluent()
+      .addDomain(DOMAIN.asString)
+      .addUser(BOB.asString(), BOB_PASSWORD)
+      .addUser(ANDRE.asString(), ANDRE_PASSWORD)
+
+    requestSpecification = baseRequestSpecBuilder(server)
+      .setAuth(authScheme(UserCredential(BOB, BOB_PASSWORD)))
+      .addHeader(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .build()
+  }
+
+  @Test
+  def shouldReturnPublicAssetCapabilityInSessionRoute(): Unit =
+    `given`()
+    .when()
+      .get("/session")
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .body("capabilities", hasKey("com:linagora:params:jmap:public:assets"))
```

Comment:
Should probably check it returns the field publicAssetTotalSize with the capability in the session?